### PR TITLE
Feature/versions script

### PR DESCRIPTION
**Pull request by pciavald**
`npm run generate-release`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "package-windows": "npm run build:electron && electron-packager . --overwrite --asar=true --platform=win32 --arch=ia32 --prune=true --out=release-builds",
     "package-linux": "npm run build:electron && electron-packager . --overwrite --asar=true --platform=linux --arch=x64 --prune=true --out=release-builds",
     "coveralls": "cat ./coverage/lcov.info | node ./node_modules/coveralls/bin/coveralls.js",
-    "compdoc": "compodoc -p tsconfig.json"
+    "compdoc": "compodoc -p tsconfig.json",
+    "generateRelease": "node ./modules/clientBinaries/generateVersions.js"
```
Comment (rynomster) [inline package.json]: could we stick to the current standards please
`generate-release`

File: modules/clientBinaries/generateVersions.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+ * Gets all hashes of current version for a specific platform
+ */
+var getHashesForPlatform = function (platform, path, hashes, promises) {
+  // this promise is resolved when both HTTP calls returned
+  return new Promise((resolve, reject) => {
+
+    got(`${signaturesURL}/${path}/${maintainer}`).then(response => {
+
+      var files = JSON.parse(response.body);
+      for (id in files) {
```
Comment (rynomster) [inline modules/clientBinaries/generateVersions.js]: I suggest using
files.forEach(file => {
})
its allot more efficient

File: modules/clientBinaries/generateVersions.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+          platforms: {}
+        }
+      }
+    }
+
+    var versions = JSON.parse(response.body);
+    var hashes = {};
+    var promises = [];
+
+    for (id in versions) {
```
Comment (rynomster) [inline modules/clientBinaries/generateVersions.js]: I'd suggest using
versions.forEach(ver => {
...
})

File: modules/clientBinaries/generateVersions.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        var platform = versions[id].name.substring(platformIndex + 1);
+        // wait for hashes to be added to our hashes array
+        promises.push(getHashesForPlatform(platform, versions[id].name, hashes));
+      }
+    }
+
+    // once we have all hashes
+    Promise.all(promises).then(function () {
+      // get asset details for each release entry
+      for (id in release.assets) {
```
Comment (rynomster) [inline modules/clientBinaries/generateVersions.js]: I'm not sure if this is an array?
arrays should use array.forEach(... => {
...
})

File: modules/clientBinaries/generateVersions.js [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        if (!release.assets.hasOwnProperty(id)) { continue ; }
+        var asset = release.assets[id];
+        var entry = getAssetDetails(asset, hashes, version);
+        if (entry) {
+          binaries.push(entry);
+        }
+      }
+      // include entries in JSON object
+      var platforms = json.clients.particld.platforms;
+      for (id in binaries) {
```
Comment (rynomster) [inline modules/clientBinaries/generateVersions.js]: we can reduce complexity using
array.forEach

File: modules/clientBinaries/generateVersions.js (hunk 2 above)
Comment (pciavald) [reply modules/clientBinaries/generateVersions.js]: you are way right

**Comment by pciavald** [comment]
wat ? :D

**Pull request merged**